### PR TITLE
feat(auth): Added key retrieval by x5t header claim

### PR DESCRIPTION
**Pull request by ahoeing**

### CODE REVIEW

**Review by JoergSiebahn — COMMENTED** [review]
Thanks for the enhancement. Could you please check the comments?

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/PublicKeyLoader.java [hunk 1]
```diff
@@ -62,7 +93,9 @@ public int getTotalNumberOfKeySources() {
   }
 
   public int getTotalNumberOfKeys() {
-    return keysWithoutKeyId.size() + keysByKid.size();
+    return Sets.newConcurrentHashSet(
+            Iterables.concat(keysByKid.values(), keysByx5t.values(), keysWithoutAnyId))
+        .size();
```
[the remark this answers is not in the document]
Comment (ahoeing) [reply sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/PublicKeyLoader.java]: done

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthService.java [hunk 2]
```diff
@@ -40,22 +40,26 @@ public AuthService(PublicKeyLoader publicKeyLoader, long leeway) {
...
+              "Verifying token without kid and x5t trying {} public keys", keysWithoutId.size());
         }
         Collections.reverse(keysWithoutId);
         return keysWithoutId.stream()
             .map(k -> verifyJwtSignature(authorizationToken, k))
             .filter(Optional::isPresent)
             .map(Optional::get)
             .findFirst()
-            .orElseThrow(() -> new JwtAuthException("Could not verify JWT without kid."))
+            .orElseThrow(() -> new JwtAuthException("Could not verify JWT without kid nor x5t."))
```
[the remark this answers is not in the document]
Comment (ahoeing) [reply sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthService.java:59]: done

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthService.java [hunk 3]
```diff
@@ -40,22 +40,26 @@ public AuthService(PublicKeyLoader publicKeyLoader, long leeway) {
   public Map<String, Claim> auth(String authorizationToken) {
     try {
       String keyId = JWT.decode(authorizationToken).getKeyId();
-      if (keyId == null) {
-        // check all keys without id
-        List<LoadedPublicKey> keysWithoutId = publicKeyLoader.getKeysWithoutId();
+      String x5t = JWT.decode(authorizationToken).getHeaderClaim("x5t").asString();
```
[the remark this answers is not in the document]
Comment (ahoeing) [reply sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthService.java]: done

**Review by JoergSiebahn — APPROVED** [review]

**Pull request merged**